src/components/input/input.scss

```diff
@@ -152,3 +153,7 @@
     -webkit-text-fill-color: color.$text-dark-muted; /* 1 */
   }
 }
+
+.c-input__label {
+  margin-bottom: ms.step(-5, 1rem);
+}
```

Comment:
This isn't a valid element because it's a sibling.

We should use an object class for this relationship. It's okay to then _use_ that object class within the component template, but as-is this is breaking our BEM conventions.

Reply:
I'm not sure what the class to use for this should be. And could I add it to this input scss file or would it need to live in its own file in src/objects?

Reply:
It feels like it might make sense for the .c-input to be applied to the label when there is a label, since it is the "root element" of the component. How about `.c-input` (root element, either input, textarea, select, or label), `.c-input__input` (input, select, or textarea), `.c-input__label` (the div containing the label text when there is a label)

Reply:
If we want to encourage usage of `c-input` in varying layouts... for example, if we think we may want to have things like joined inputs or inputs where the label is to the side or something like that... then creating a new pattern like `o-form-group` ([example](https://primer.style/css/components/forms#form-groups)) is probably the right approach. That's my instinct.

Alternatively, we could refactor `c-input` so that it's structured more like this:

```html
<div class="c-input">
  <label class="c-input__label">...</label>
  <input class="c-input__control" ... />
</div>
```

But as I'm typing that out, I'm doubting myself a bit on that. 🤔

Reply:
I think I'm in favor of whichever solution is least-resistance to getting this PR merged. I think that creating a new pattern is a little out of scope for this PR.

In terms of restructuring `c-input`, I think it makes more sense to associate the label and form element by wrapping the label around the input, because dealing with id's is more complicated and it is easy to end up making mistakes with duplication. Here's what I suggested in my last comment, as a code block:

```html
<label class="c-input">
  <div class="c-input__label">...</div>
  <input class="c-input__control" ... />
</label>
```

Reply:
> I think that creating a new pattern is a little out of scope for this PR.

Maybe, but in that case shouldn't we create the smaller pattern dependency as a separate PR and then update this one?

It seems like what we realize we need is this sort of structure:

```
c-comment-form
├── o-form-group
│   └── c-input
├── o-form-group
│   └── c-input
└── (etc.)
```

We've merged in `c-input`, and you've created a PR for `c-comment-form`. But we need `o-form-group`. I don't understand why we wouldn't go create `o-form-group` instead of refactoring `c-input`?

Reply:
Can you open an issue for o-form-group with examples of the use-cases?

Reply:
I sure can! 👍

Reply:
Done: #1258